common/pkg/capi/clustertopology/handlers/cluster_context.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package handlers
+
+import (
+	"context"
+	"errors"
+
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// clusterKeyContextKey is how we find a cluster key from a context.Context.
+type clusterKeyContextKey struct{}
+
+func ClusterKeyFrom(ctx context.Context) (client.ObjectKey, error) {
+	clusterKey, ok := ctx.Value(clusterKeyContextKey{}).(client.ObjectKey)
+	if !ok || clusterKey.Name == "" {
+		return client.ObjectKey{}, errors.New(
+			"failed to detect cluster name from GeneratePatch request",
+		)
+	}
+
+	return clusterKey, nil
+}
+
+func ClusterKeyInto(
```

Comment:
I was a little confused by the function name and the signature that it takes in a context and then returns a context, is there a more descriptive function name?

```suggestion
func SetClusterKeyInto(
```